isis/src/base/objs/LinearAlgebra/LinearAlgebra.cpp

```diff
@@ -343,36 +346,32 @@ namespace Isis {
 
 
   /**
-   * Returns the psuedoinverse of a matrix. Throws an error if the given matrix
-   * is not left or right invertible.
-   *
-   * @param matrix The matrix to compute the psuedoinverse of.
+   * Returns the pseudoinverse of a matrix.
    *
-   * @return @b LinearAlgebra::Matrix The psuedoinverse matrix.
+   * @param matrix The matrix to compute the pseudoinverse of.
    *
-   * @throw IException::Programmer "Unable to invert the given matrix."
+   * @return @b LinearAlgebra::Matrix The pseudoinverse matrix.
    */
-  LinearAlgebra::Matrix LinearAlgebra::psuedoinverse(const Matrix &matrix) {
-    try {
-      // For square matrices, the psuedoinverse is just the inverse
-      if (matrix.size1() == matrix.size2()) {
-        return inverse(matrix);
-      }
-      // If there are more rows than columns compute a left inverse
-      // (M^T * M)^-1 * M^T
-      else if (matrix.size1() > matrix.size2()) {
-        return multiply(inverse(multiply(transpose(matrix), matrix)), transpose(matrix));
-      }
-      // Otherwise, more columns than rows, compute a right inverse
-      // M^T * (M * M^T)^-1
-      else {
-        return multiply(transpose(matrix), inverse(multiply(matrix, transpose(matrix))));
+  LinearAlgebra::Matrix LinearAlgebra::pseudoinverse(const Matrix &matrix) {
+    // Copy values into Armadillo matrix
+    arma::mat arMat(matrix.size1(), matrix.size2());
+    for (int i = 0; i < matrix.size1(); i++) {
+      for (int j = 0; j < matrix.size2(); j++) {
+        arMat(i, j) = matrix(i, j);
```

Comment:
There is likely an easier way to do these data copies with just accessing the raw data array but the boost storage format can be finicky and I don't feel like figuring out the code needed to keep it in line with Armadillo.